package.json

```diff
@@ -38,6 +38,7 @@
     "@popperjs/core": "^2.2.2",
     "@types/puppeteer": "^2.0.1",
     "@types/uuid": "^7.0.3",
+    "amazon-chime-sdk-js": "^2.4.0",
```

Comment:
The [AudioVideoProvider Storybook](https://aws.github.io/amazon-chime-sdk-component-library-react/?path=/story/sdk-providers-audiovideoprovider--page) has an example of importing both JS SDK and the component library. A builder imports `DefaultMeetingSession` from JS SDK, instantiates, and passes it into `AudioVideoContext.Provider`.

```
import { AudioVideoContext } from 'amazon-chime-sdk-component-library-react';
import { DefaultMeetingSession } from 'amazon-chime-sdk-js';
```

I think we need to explain how to achieve this example in a new repo setup where a peer dependency is no more there.

Reply:
Does this example change? I thought Richard mentioned that builders can still access the transitive dependency directly

Otherwise, I would expect our current repo demo to break since it relies on the [SDK directly for the TestSound](https://github.com/aws/amazon-chime-sdk-component-library-react/blob/master/demo/meeting/src/utils/TestSound.tsx#L7)

Reply:
Cool. We just need an explanation for builders.

Reply:
We need to be in one of two positions:

* Customers install their own version of the SDK, it's a peer dependency of the component library, and the versions must match. They should use a broad caret, and we should tighten it in a peer dependency relationship.
* Customers should not install their own version of the SDK, and it should only be provided implicitly by the SDK.

The situation we want to avoid is that the module tree has two different versions of the SDK installed:

```
$ npm ls --all | fgrep amazon
├─┬ amazon-chime-sdk-component-library-react@1.6.1
│ ├── amazon-chime-sdk-js@2.4.0
└─┬ amazon-chime-sdk-js@1.22.0
```

Reply:
@ltrung in your testing were you able to get any conflicting versions with this approach? What happened for the three test cases we discussed?

Reply:
If JS SDK is a direct dependency in component library:
Case 1: If the package consumes component library does not have any existing dependency on JS SDK, installing component library would automatically pull in JS SDK with the correct version (v2.4.1).
Case 2: If the package consumes component library and has an existing dependency on JS SDK ^2.0.0, installing component library would automatically pull in JS SDK v2.4.1.
Case 3: If the package consumes component library and has an existing dependency on JS SDK ^1.x, installing component library would result in 2 different of JS SDK (v2.4.1 and v1.22.0)

If we also specify JS SDK as a peer dependency in component library, NPM gave a warning about peer dependency not met but still install both versions.

If we specify JS SDK as dev and peer dependency in component library, in all three cases, consumer packages would need to make sure the current installed version of JS SDK is compatible with component library. If not, npm would give a warning but does not auto download the compatible JS SDK.

Reply:
From the test results above, to avoid the case where the module tree contains 2 different version of JS SDK, we will still use devDependency and peerDependency and customers would need to install their own version of JS SDK and make sure it matches the component library.

Reply:
And if we set it up with dev and peer dependencies, the three cases are:

* The customer manually installs a conflicting version. What happens? (I expect: error or warning about a peer dependency, but we need this to fail somehow.)
* The customer does not install anything. What happens? (I expect: build bustage.)
* The customer installs the right version.

Reply:
- If the customer installs the right version, npm would not get any error or warning.
- If the customer does not install anything, npm install would give a warning that the JS SDK version that component library requires is not installed and customers would need to install it themselves.
- If the customer installed a conflicting version of JS SDK, npm would install that version but also give a warning that the JS SDK version that component library requires is not installed and customers would need to install it themselves.

Reply:
OK, so we should do peer dep and explore what options we have for detecting conflicts more robustly here.